features/384-k8s-access/index.md

```diff
@@ -0,0 +1,61 @@
+# Access to k8s
+
+Instead of wrapping k8s APIs with skycli, we can generate a token to allow limited access to
+k8s resources.
+
+# Benefits
+
+The developer can use kubectl and connect to the k8s API server directly.
+kubectl is a very mature program and is known by many developers.
+
+We can also avoid reimplementing features of kubectl.
+
+# Authentication strategies of k8s
+
+k8s supports the following authentication strategies.
+
+- [Client Certificate](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs)
+- [Static Bearer token](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#static-token-file)
+- [HTTP basic auth](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#static-password-file)
+- [Service Account](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#service-account-tokens)
+- [OpenID Connect](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens)
+- [Authentication webhook](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#webhook-token-authentication)
+- [Authenticating proxy](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#authenticating-proxy)
+
+The most common one is Service Account.
+It is supported natively thus supported by all providers.
+Therefore it is the chosen one.
+
+# Approach
+
+We define a ClusterRole to allow read access to essential resources.
```

Comment:
I think ClusterRole allows access to all resources in the cluster? i.e. pods in other apps can be accessed too.

Reply:
No. The resolution is still bound by the RoleBinding. Since RoleBinding belongs to a namespace. The access is scoped within that namespace. The benefit of using ClusterRole is to avoid creating one role per namespace.